File: calibration_manager/sensor/wheel.py
import yaml
import numpy as np

from calibration_manager.sensor.sensor import Sensor


class Wheel(Sensor):
    def __init__(self):
        super(Wheel, self).__init__()
        self.diameter = 0
        self.scale = 1.0

    def read_intrinsic_calibration(self, path):
        with open(path, 'r') as f:
            data = yaml.safe_load(f)
            self.diameter = float(data.get('diameter', 0))
            self.scale = float(data.get('scale', 1.0))

File: src/calibration_manager/Wheel.cpp
#include <sstream>
#include <yaml-cpp/yaml.h>

#include "calibration_manager/Wheel.h"

namespace octopus {
namespace calibration_manager {

void Wheel::ReadInstrinsicParameters(const std::string& path) {
    YAML::Node node = YAML::LoadFile(path);

    //set wheel intrinsic
    diameter = node["diameter"].as<double>();
    scale = node["scale"].as<double>();
}

std::string Wheel::ToString() const {
    std::stringstream result;
    result << "Wheel Configuration: " << std::endl;
    result << "diameter = " << this->diameter << std::endl;
    result << "scale = " << this->scale << std::endl;
    return result.str();
}

} //namespace calibration_manager
} // namespace octopus